xarray/tests/test_indexes.py

```diff
@@ -0,0 +1,63 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from xarray.core.indexes import PandasIndex, PandasMultiIndex, _asarray_tuplesafe
+
+
+def test_asarray_tuplesafe():
+    res = _asarray_tuplesafe(("a", 1))
+    assert isinstance(res, np.ndarray)
+    assert res.ndim == 0
+    assert res.item() == ("a", 1)
+
+    res = _asarray_tuplesafe([(0,), (1,)])
+    assert res.shape == (2,)
+    assert res[0] == (0,)
+    assert res[1] == (1,)
+
+
+class TestPandasIndex:
+    def test_query(self):
+        # TODO: add tests that aren't just for edge cases
+        index = PandasIndex(pd.Index([1, 2, 3]))
```

Comment:
Forgive the tweaky point, but should we call this something that indicates it's our wrapper?

Reply:
We've discussed this in #5102 and came up with `Index` and `PandasIndex` (see https://github.com/pydata/xarray/pull/5102#issuecomment-813090844 and https://github.com/pydata/xarray/pull/5102#discussion_r624341647). I'm still open to discussion, though! I don't have strong opinion on this.

Reply:
Great